client/src/js/base/__tests__/InputSave.test.js

```diff
@@ -19,312 +25,81 @@ describe("<InputSave />", () => {
-
-            mockEvent = {
-                relatedTarget: {
-                    type: "submit"
-                }
-            };
-
-            expect(spyAlternate).not.toHaveBeenCalled();
-            expect(() => {
-                wrapper.find(FormControl).simulate("blur", mockEvent);
-            }).not.toThrow();
-            expect(spyAlternate).toHaveBeenCalled();
-
-            spyAlternate.mockReset();
-            spyAlternate.mockRestore();
-        });
+    it("handleChange should not update state when [this.props.disabled=true]]", () => {
+        props.disabled = true;
+        const wrapper = shallow(<InputSave {...props} />);
+        wrapper.find("FormControl").simulate("change", e);
+        expect(wrapper.state()).toEqual({ pending: false, value: 5 });
     });
 
-    describe("Button subcomponent, handleSubmit:", () => {
-        let spyHandler;
-        let spyBlur;
-
-        beforeEach(() => {
-            props = {
-                onSave: jest.fn(),
-                initialValue: "initial_value"
-            };
-            wrapper = mount(<InputSave {...props} />);
-
-            spyHandler = sinon.spy(wrapper.instance(), "handleSubmit");
-            spyBlur = sinon.spy(wrapper.instance(), "blur");
-
-            wrapper.instance().forceUpdate();
-        });
-
-        afterEach(() => {
-            spyHandler.restore();
-            spyBlur.restore();
-        });
-
-        it("when there is no change in value, click/enter triggers submit handler and blur", () => {
-            wrapper.find(Button).simulate("submit");
-
-            expect(spyHandler.calledOnce).toBe(true);
-            expect(spyBlur.calledOnce).toBe(true);
+    it("handleSubmit should call props.onSave when [this.state.value!=this.props.initialValue]", () => {
+        const wrapper = mount(<InputSave {...props} />);
+        wrapper.setState({ ...state, value: 10 });
+        wrapper.find("FormControl").simulate("submit", e);
+        expect(props.onSave).toHaveBeenCalledWith({
+            max: 10,
+            min: 0,
+            name: "test_inputsave",
+            value: 10
         });
+    });
 
-        it("when there is a change in value, click/enter sets [state.pending=true] and calls the onSave prop", () => {
-            wrapper.setState({ value: "CHANGED" });
-            wrapper.find(Button).simulate("submit");
+    it("handleSubmit should not call props.onSave when [this.state.value=this.props.initialValue]", () => {
+        const wrapper = mount(<InputSave {...props} />);
+        wrapper.find("FormControl").simulate("submit", e);
+        expect(props.onSave).not.toHaveBeenCalled();
+    });
 
-            expect(spyHandler.calledOnce).toBe(true);
-            expect(wrapper.state("pending")).toEqual(true);
-            expect(props.onSave).toHaveBeenCalled();
-        });
+    it("handleSubmit should not call props.onSave when [this.state.value=this.props.initialValue]", () => {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Test title is used multiple times in the same describe block.](https://app.codacy.com/manual/virtool/virtool/pullRequest?prid=4770288)